src/Core/Content/Seo/Api/SeoActionController.php

```diff
@@ -197,15 +201,21 @@ public function createCustomSeoUrls(RequestDataBag $dataBag, Context $context):
         }
 
         foreach ($writeData as $salesChannelId => $writeRows) {
-            $salesChannelEntity = null;
-
             if ($salesChannelId === '') {
                 throw SeoException::salesChannelIdParameterIsMissing();
             }
 
             /** @var SalesChannelEntity $salesChannelEntity */
             $salesChannelEntity = $salesChannels->get($salesChannelId);
 
+            if ($salesChannelEntity === null) {
+                throw SeoException::salesChannelNotFound($salesChannelId);
+            }
+
+            if ($salesChannelEntity->getTypeId() === Defaults::SALES_CHANNEL_TYPE_API) {
```

Comment:
not sure if I should throw an exception here or skip it. or not adjust the controller here at all.